### ISSUE TITLE
Suitable content access module?

### ISSUE THREAD

**Issue by Graham-72** [backdrop-ops/contrib#105]
Can anyone please recommend a suitable module to port to Backdrop (if necessary) that enables me to grant persmissions on a per node basis? I would like to provide a 'sandbox' page within a site that can be edited by any authenticated user.

How about the Drupal modules 'node access' or 'content access'? I have no experience of either of them. Anything else?

**Comment by klonos** [comment]
A little bit of a stats summary has always helped me decide in similar cases...

[Node Access](https://www.drupal.org/project/node_access) that seems to be deprecated ([less than half a thousand reported installs and declining](https://www.drupal.org/project/usage/node_access)) over [Nodeaccess](https://www.drupal.org/project/nodeaccess) that currently counts a bit over 10k installs, states in its project page over at d.org (for Nodeaccess):

> Long standing node access security module, as with content access at the time that this module (node_access) was being developed the nodeaccess module did not have a Drupal 6 version and repeated attempts to contact the authors and or moderators had failed.

...furthermore, they seem to prefer [Content Access](https://www.drupal.org/project/content_access) over Nodeaccess and have a "(recommended)" next to their mention of it.

[Content Access](https://www.drupal.org/project/content_access) has over 50k installs, which on its own should say enough (plus they'd be more candidates for "converting" to BD :smiling_imp:). Although, to be fair, we should mention that Nodeaccess is in *Actively maintained
/Under active development* status, while Content Access is *Seeking co-maintainer(s)/Maintenance fixes only*.

Personally, I like Content Access. It hasn't failed me when I needed something similar in the past.

Hope this helps.

**Comment by quicksketch** [comment]
Ultimately, I find that if a site needs access control, the scope keeps expanding until [OG](https://www.drupal.org/project/og) is the most suitable tool. Even though it's the most complicated, it's also one of the best maintained and written.

Optionally, if this is a *very* small use-case, perhaps just create a new content type and make a single node of that type? Then grant authenticated users access to "edit any" of that type, but not create new ones.

**Comment by Graham-72** [comment]
Thanks for the suggestions. My immediate need is a very small-use case, so I will create a new content type. Sounds an excellent plan.

**Comment by sentaidigital** [comment]
I'm looking for something similar, but for granting read access, essentially premium content.  Role X may see nodes of Content Type Y is sufficient for my use case.  I'm pretty sure any of the ones @klonos mentioned can do this, but have any of them been ported or is anyone working on porting one?  I'd rather not duplicate efforts or spend a lot of time on one if another is nearly ready.

**Comment by klonos** [comment]
@sentaidigital you can search [here](https://github.com/backdrop-contrib). Searching for "access" yields these results for me:

[simple_access_grants](https://github.com/backdrop-contrib/simple_access_grants)

[simple_access](https://github.com/backdrop-contrib/simple_access)

[content_access](https://github.com/backdrop-contrib/content_access)

[access_unpublished](https://github.com/backdrop-contrib/access_unpublished)

**Comment by sentaidigital** [comment]
Hmmm... I did a bunch of searches [here](https://backdropcms.org/modules) before posting, but either I had such generic terms that everything was returned, or too many terms and got back nothing.  Searching for [content access](https://backdropcms.org/modules?s=content+access&url=) returned nothing at all.

**Comment by klonos** [comment]
Yeah, the [content_access repo](https://github.com/backdrop-contrib/content_access) seems to be empty(?), but mind you, the module/theme/layout search on backdropcms.org only works if the projects have actual releases on GitHub (and the Project Installer from within your website in turn pulls whatever backdropcms.org has available). So, access_unpublished will not show up because [it has no releases](https://github.com/backdrop-contrib/access_unpublished/releases) (at the time of writing).

**Comment by klonos** [comment]
...plus the search at backdropcms.org seems to be have some glitches: https://github.com/backdrop-ops/backdropcms.org/issues/232

**Comment by Graham-72** [comment]
@sentaidigital After looking at these modules I produced and use [simple_access_grants](https://github.com/backdrop-contrib/simple_access_grants) to meet my need for a 'members only' section in a site.

**Comment by laryn** [comment]
For anyone coming to this thread in the future:

 - [Content Access](https://github.com/backdrop-contrib/content_access) is near a full release 
 - [Content View Access](https://github.com/backdrop-contrib/content_view_access) is another option that I've used on a number of sites for simple "Members' Only" content by content type.
